File: backend/app/schemas/outreach.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.models.enums import (
    OutreachCampaignStatus,
    OutreachChannel,
    OutreachConversationStatus,
    OutreachMessageRole,
    QualificationVerdict,
)


class OutreachCampaignCreate(BaseModel):
    channel: OutreachChannel = OutreachChannel.internal
    subject: str | None = Field(default=None, max_length=500)
    recipient_email: EmailStr | None = None
    recipient_phone: str | None = Field(default=None, max_length=50)


class LeadReplyCreate(BaseModel):
    content: str = Field(min_length=1)
    auto_vet: bool = True


class OutreachCampaignResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    lead_id: UUID
    deployment_package_id: UUID | None
    created_by_id: UUID | None
    status: OutreachCampaignStatus
    channel: OutreachChannel
    subject: str | None
    recipient_email: str | None
    recipient_phone: str | None
    agent_metadata: dict[str, Any]
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class OutreachConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    campaign_id: UUID
    lead_id: UUID
    status: OutreachConversationStatus
    qualification_score: int | None
    qualification_verdict: QualificationVerdict | None
    vetting_result: dict[str, Any] | None
    summary: str | None
    created_at: datetime
    updated_at: datetime


class OutreachMessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    conversation_id: UUID
    role: OutreachMessageRole
    channel: OutreachChannel
    subject: str | None
    content: str
    delivery_metadata: dict[str, Any]
    created_at: datetime
    updated_at: datetime


class PaginatedOutreachCampaignsResponse(BaseModel):
    items: list[OutreachCampaignResponse]
    total: int
    limit: int
    offset: int


class PaginatedOutreachMessagesResponse(BaseModel):
    items: list[OutreachMessageResponse]
    total: int
    limit: int
    offset: int


class VettingResponse(BaseModel):
    conversation: OutreachConversationResponse
    qualification_score: int
    qualification_verdict: QualificationVerdict
    summary: str
    recommended_next_step: str
    buying_signals: list[str]
    red_flags: list[str]

File: frontend/src/types/outreach.ts
export type OutreachCampaignStatus =
  | 'draft'
  | 'active'
  | 'paused'
  | 'completed'
  | 'failed'

export type OutreachChannel = 'internal' | 'email' | 'sms' | 'voice'

export type OutreachConversationStatus =
  | 'open'
  | 'qualified'
  | 'disqualified'
  | 'handoff'
  | 'closed'

export type OutreachMessageRole = 'agent' | 'lead' | 'system'

export type QualificationVerdict =
  | 'qualified'
  | 'needs_more_info'
  | 'disqualified'
  | 'handoff'

export interface OutreachCampaign {
  id: string
  organization_id: string
  lead_id: string
  deployment_package_id: string | null
  created_by_id: string | null
  status: OutreachCampaignStatus
  channel: OutreachChannel
  subject: string | null
  recipient_email: string | null
  recipient_phone: string | null
  agent_metadata: Record<string, unknown>
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface OutreachConversation {
  id: string
  organization_id: string
  campaign_id: string
  lead_id: string
  status: OutreachConversationStatus
  qualification_score: number | null
  qualification_verdict: QualificationVerdict | null
  vetting_result: Record<string, unknown> | null
  summary: string | null
  created_at: string
  updated_at: string
}

export interface OutreachMessage {
  id: string
  organization_id: string
  conversation_id: string
  role: OutreachMessageRole
  channel: OutreachChannel
  subject: string | null
  content: string
  delivery_metadata: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface PaginatedOutreachCampaignsResponse {
  items: OutreachCampaign[]
  total: number
  limit: number
  offset: number
}

export interface PaginatedOutreachMessagesResponse {
  items: OutreachMessage[]
  total: number
  limit: number
  offset: number
}

export interface OutreachCampaignCreateRequest {
  channel?: OutreachChannel
  subject?: string | null
  recipient_email?: string | null
  recipient_phone?: string | null
}

export interface OutreachCampaignListParams {
  lead_id?: string
  status?: OutreachCampaignStatus
  limit?: number
  offset?: number
}

export type OutreachCampaignSortField =
  | 'subject'
  | 'channel'
  | 'status'
  | 'created_at'

export type SortDirection = 'asc' | 'desc'

export interface LeadReplyCreateRequest {
  content: string
  auto_vet?: boolean
}

export interface VettingResponse {
  conversation: OutreachConversation
  qualification_score: number
  qualification_verdict: QualificationVerdict
  summary: string
  recommended_next_step: string
  buying_signals: string[]
  red_flags: string[]
}

export interface VettingInsights {
  qualification_score: number | null
  qualification_verdict: QualificationVerdict | null
  summary: string | null
  recommended_next_step: string | null
  buying_signals: string[]
  red_flags: string[]
}
